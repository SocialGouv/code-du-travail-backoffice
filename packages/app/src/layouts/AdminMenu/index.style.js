import styled from "@emotion/styled";

import _Subtitle from "../../elements/Subtitle";

export const Container = styled.div`
  background-color: var(--color-dark-slate-gray);
  display: flex;
  flex-direction: column;
  min-width: 13rem;
  overflow-y: auto;
  width: 13rem;
`;

export const Subtitle = styled(_Subtitle)`
  color: white;
  padding: 0 0.5rem;
`;

export const Link = styled.button`
  appearance: none;
  background-color: ${p => (p.isActive ? "black" : "transparent")};
  border: 0;
  border-bottom: 1px solid rgba(255, 255, 255, 0.1);
  color: ${p => (p.isActive ? "white" : "rgba(255, 255, 255, 0.75)")};
  cursor: pointer;
  font-family: inherit;
  font-size: inherit;
  font-weight: inherit;
  padding: 0.5rem;
  text-align: left;

  :hover {
    background-color: ${p => (p.isActive ? "black" : "var(--color-japenese-indigo)")};
    color: white;
  }
`;
