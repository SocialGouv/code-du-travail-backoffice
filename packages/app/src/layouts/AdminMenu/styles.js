import styled from "@emotion/styled";
import css from "styled-jsx/css";

import _Subtitle from "../../elements/Subtitle";

export default css`
  .Container {
    background-color: var(--color-dark-slate-gray);
    display: flex;
    flex-direction: column;
    min-width: 13rem;
    overflow-y: auto;
    width: 13rem;
  }
`;

export const Subtitle = styled(_Subtitle)`
  color: white;
  padding: 0 0.5rem;
`;

export const Link = styled.div`
  background-color: ${p => (p.isActive ? "black" : "transparent")};
  border-bottom: 1px solid rgba(255, 255, 255, 0.1);
  color: ${p => (p.isActive ? "white" : "rgba(255, 255, 255, 0.75)")};
  cursor: pointer;
  padding: 0.5rem;

  :hover {
    background-color: ${p => (p.isActive ? "black" : "var(--color-japenese-indigo)")};
    color: white;
  }
`;
