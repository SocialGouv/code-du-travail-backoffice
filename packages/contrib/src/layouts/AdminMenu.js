import Router from "next/router";
import React from "react";
import { Flex } from "rebass";
import styled from "styled-components";

import _Subtitle from "../elements/Subtitle";

const Container = styled(Flex)`
  background-color: var(--color-dark-slate-gray);
  min-width: 12rem;
  width: 12rem;
`;
const Subtitle = styled(_Subtitle)`
  color: white;
  padding: 0 0.5rem;
`;

const Link = styled.div`
  background-color: ${props =>
    Boolean(props.isCurrent) ? "black" : "transparent"};
  border-bottom: 1px solid rgba(255, 255, 255, 0.1);
  color: ${props =>
    Boolean(props.isCurrent) ? "white" : "rgba(255, 255, 255, 0.75)"};
  cursor: pointer;
  padding: 0.5rem;

  :hover {
    background-color: ${props =>
      Boolean(props.isCurrent) ? "black" : "var(--color-japenese-indigo)"};
    color: white;
  }
`;

export default class AdminMenu extends React.PureComponent {
  goTo(resource) {
    Router.push(`/admin${resource}`);
  }

  isCurrent(path) {
    return path.length !== 0
      ? window.location.pathname.startsWith(`/admin${path}`)
      : window.location.pathname === "/admin";
  }

  render() {
    return (
      <Container flexDirection="column">
        <Link isCurrent={this.isCurrent("")} onClick={() => this.goTo("")}>
          Tableau de bord
        </Link>
        <Link
          isCurrent={this.isCurrent("/agreements")}
          onClick={() => this.goTo("/agreements")}
        >
          Conventions
        </Link>
        <Link
          isCurrent={this.isCurrent("/tags")}
          onClick={() => this.goTo("/tags")}
        >
          Étiquettes
        </Link>
        <Link
          isCurrent={this.isCurrent("/questions")}
          onClick={() => this.goTo("/questions")}
        >
          Questions
        </Link>
        <Link
          isCurrent={this.isCurrent("/answers")}
          onClick={() => this.goTo("/answers")}
        >
          Réponses
        </Link>
        <Link
          isCurrent={this.isCurrent("/locations")}
          onClick={() => this.goTo("/locations")}
        >
          Unités
        </Link>
        <Link
          isCurrent={this.isCurrent("/users")}
          onClick={() => this.goTo("/users")}
        >
          Utilisateurs
        </Link>

        <Subtitle>Maintenance</Subtitle>
        <Link
          isCurrent={this.isCurrent("/migrations")}
          onClick={() => this.goTo("/migrations")}
        >
          Migrations
        </Link>
      </Container>
    );
  }
}
