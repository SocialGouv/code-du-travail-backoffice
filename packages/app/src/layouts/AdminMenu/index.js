import Router from "next/router";
import React from "react";

import cache from "../../cache";
import { USER_ROLE } from "../../constants";
import styles, { Link, Subtitle } from "./styles";

export default class AdminMenu extends React.PureComponent {
  goTo(resource) {
    Router.push(`/admin${resource}`);
  }

  render() {
    const me = cache.get("me");

    if (me === null) {
      return (
        <div className="Container">
          <style jsx>{styles}</style>
        </div>
      );
    }

    const isRegionalAdmin = me.role === USER_ROLE.REGIONAL_ADMINISTRATOR;

    if (isRegionalAdmin) {
      return (
        <div className="Container">
          <style jsx>{styles}</style>
          <Link
            onClick={() => this.goTo("")}
            onKeyPress={() => this.goTo("")}
            role="link"
            tabIndex="0"
          >
            Tableau de bord
          </Link>
        </div>
      );
    }

    return (
      <div className="Container">
        <style jsx>{styles}</style>
        <Link
          onClick={() => this.goTo("")}
          onKeyPress={() => this.goTo("")}
          role="link"
          tabIndex="0"
        >
          Tableau de bord
        </Link>
        <Link
          onClick={() => this.goTo("/agreements")}
          onKeyPress={() => this.goTo("/agreements")}
          role="link"
          tabIndex="0"
        >
          Conventions
        </Link>
        <Link
          onClick={() => this.goTo("/tags")}
          onKeyPress={() => this.goTo("/tags")}
          role="link"
          tabIndex="0"
        >
          Étiquettes
        </Link>
        <Link
          onClick={() => this.goTo("/tags-categories")}
          onKeyPress={() => this.goTo("/tags-categories")}
          role="link"
          tabIndex="0"
        >
          └ Catégories
        </Link>
        <Link
          onClick={() => this.goTo("/questions")}
          onKeyPress={() => this.goTo("/questions")}
          role="link"
          tabIndex="0"
        >
          Questions
        </Link>
        <Link
          onClick={() => this.goTo("/answers")}
          onKeyPress={() => this.goTo("/answers")}
          role="link"
          tabIndex="0"
        >
          Réponses
        </Link>
        <Link
          onClick={() => this.goTo("/generic-answers")}
          onKeyPress={() => this.goTo("/generic-answers")}
          role="link"
          tabIndex="0"
        >
          Réponses génériques
        </Link>
        <Link
          onClick={() => this.goTo("/locations")}
          onKeyPress={() => this.goTo("/locations")}
          role="link"
          tabIndex="0"
        >
          Unités
        </Link>
        <Link
          onClick={() => this.goTo("/users")}
          onKeyPress={() => this.goTo("/users")}
          role="link"
          tabIndex="0"
        >
          Utilisateurs
        </Link>
        <Link
          onClick={() => this.goTo("/areas")}
          onKeyPress={() => this.goTo("/areas")}
          role="link"
          tabIndex="0"
        >
          Zones
        </Link>

        <Subtitle>Data Filler</Subtitle>
        <Link
          onClick={() => this.goTo("/themes")}
          onKeyPress={() => this.goTo("/themes")}
          role="link"
          tabIndex="0"
        >
          Thèmes
        </Link>
        <Link
          onClick={() => this.goTo("/requests")}
          onKeyPress={() => this.goTo("/requests")}
          role="link"
          tabIndex="0"
        >
          Requêtes
        </Link>
        <Link
          onClick={() => this.goTo("/terms")}
          onKeyPress={() => this.goTo("/terms")}
          role="link"
          tabIndex="0"
        >
          Glossaire
        </Link>

        <Subtitle>Maintenance</Subtitle>
        <Link
          onClick={() => this.goTo("/data-filler")}
          onKeyPress={() => this.goTo("/data-filler")}
          role="link"
          tabIndex="0"
        >
          Data Filler
        </Link>
        <Link
          onClick={() => this.goTo("/logs")}
          onKeyPress={() => this.goTo("/logs")}
          role="link"
          tabIndex="0"
        >
          Logs
        </Link>
        <Link
          onClick={() => this.goTo("/migrations")}
          onKeyPress={() => this.goTo("/migrations")}
          role="link"
          tabIndex="0"
        >
          Migrations
        </Link>
      </div>
    );
  }
}
