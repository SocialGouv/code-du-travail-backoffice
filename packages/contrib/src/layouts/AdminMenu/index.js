import Router from "next/router";
import React from "react";

import cache from "../../cache";
import { USER_ROLE } from "../../constants";

import styles, { Link, Subtitle } from "./styles";

export default class AdminMenu extends React.PureComponent {
  goTo(resource) {
    Router.push(`/admin${resource}`);
  }

  isActive() {
    return false;
    // const { pathname } = window.location;
    // const regexp =
    //   path.length > 0 ? new RegExp(`^/admin${path}(?=/|$)`) : /^\/admin$/;

    // return regexp.test(pathname);
  }

  render() {
    const me = cache.get("me");

    if (me === null) {
      return (
        <div className="Container">
          <style jsx>{styles}</style>
        </div>
      );
    }

    const isRegionalAdmin = me.role === USER_ROLE.REGIONAL_ADMINISTRATOR;

    if (isRegionalAdmin) {
      return (
        <div className="Container">
          <style jsx>{styles}</style>
          <Link
            isActive={this.isActive("")}
            onClick={() => this.goTo("")}
            onKeyPress={() => this.goTo("")}
            role="link"
            tabIndex="0"
          >
            Tableau de bord
          </Link>
        </div>
      );
    }

    return (
      <div className="Container">
        <style jsx>{styles}</style>
        <Link
          isActive={this.isActive("")}
          onClick={() => this.goTo("")}
          onKeyPress={() => this.goTo("")}
          role="link"
          tabIndex="0"
        >
          Tableau de bord
        </Link>
        <Link
          isActive={this.isActive("/agreements")}
          onClick={() => this.goTo("/agreements")}
          onKeyPress={() => this.goTo("/agreements")}
          role="link"
          tabIndex="0"
        >
          Conventions
        </Link>
        <Link
          isActive={this.isActive("/tags")}
          onClick={() => this.goTo("/tags")}
          onKeyPress={() => this.goTo("/tags")}
          role="link"
          tabIndex="0"
        >
          Étiquettes
        </Link>
        <Link
          isActive={this.isActive("/tags-categories")}
          onClick={() => this.goTo("/tags-categories")}
          onKeyPress={() => this.goTo("/tags-categories")}
          role="link"
          tabIndex="0"
        >
          └ Catégories
        </Link>
        <Link
          isActive={this.isActive("/questions")}
          onClick={() => this.goTo("/questions")}
          onKeyPress={() => this.goTo("/questions")}
          role="link"
          tabIndex="0"
        >
          Questions
        </Link>
        <Link
          isActive={this.isActive("/answers")}
          onClick={() => this.goTo("/answers")}
          onKeyPress={() => this.goTo("/answers")}
          role="link"
          tabIndex="0"
        >
          Réponses
        </Link>
        <Link
          isActive={this.isActive("/generic-answers")}
          onClick={() => this.goTo("/generic-answers")}
          onKeyPress={() => this.goTo("/generic-answers")}
          role="link"
          tabIndex="0"
        >
          Réponses génériques
        </Link>
        <Link
          isActive={this.isActive("/locations")}
          onClick={() => this.goTo("/locations")}
          onKeyPress={() => this.goTo("/locations")}
          role="link"
          tabIndex="0"
        >
          Unités
        </Link>
        <Link
          isActive={this.isActive("/users")}
          onClick={() => this.goTo("/users")}
          onKeyPress={() => this.goTo("/users")}
          role="link"
          tabIndex="0"
        >
          Utilisateurs
        </Link>
        <Link
          isActive={this.isActive("/areas")}
          onClick={() => this.goTo("/areas")}
          onKeyPress={() => this.goTo("/areas")}
          role="link"
          tabIndex="0"
        >
          Zones
        </Link>

        <Subtitle>Maintenance</Subtitle>
        <Link
          isActive={this.isActive("/data-filler")}
          onClick={() => this.goTo("/data-filler")}
          onKeyPress={() => this.goTo("/data-filler")}
          role="link"
          tabIndex="0"
        >
          Data Filler
        </Link>
        <Link
          isActive={this.isActive("/logs")}
          onClick={() => (window.location.href = "/admin/logs")}
          onKeyPress={() => this.goTo("/logs")}
          role="link"
          tabIndex="0"
        >
          Logs
        </Link>
        <Link
          isActive={this.isActive("/migrations")}
          onClick={() => this.goTo("/migrations")}
          onKeyPress={() => this.goTo("/migrations")}
          role="link"
          tabIndex="0"
        >
          Migrations
        </Link>
      </div>
    );
  }
}
