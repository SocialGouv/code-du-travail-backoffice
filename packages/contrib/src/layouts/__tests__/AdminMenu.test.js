import React from "react";
import { render } from "react-testing-library";
import AdminMenu from "../AdminMenu";

describe("[Contrib] layouts/<AdminMenu />", () => {
  it("should render", () => {
    const { container } = render(<AdminMenu />);

    expect(container).toMatchSnapshot();
  });
});
