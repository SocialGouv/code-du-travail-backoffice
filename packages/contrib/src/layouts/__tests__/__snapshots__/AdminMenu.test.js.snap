// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Contrib] layouts/<AdminMenu /> should match snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa sc-bwzfXH AdminMenu__Container-t0dk93-0 dYccBZ"
  >
    <div
      class="AdminMenu__Link-t0dk93-2 LnAQE"
    >
      Tableau de bord
    </div>
    <div
      class="AdminMenu__Link-t0dk93-2 kospmv"
    >
      Conventions
    </div>
    <div
      class="AdminMenu__Link-t0dk93-2 kospmv"
    >
      Étiquettes
    </div>
    <div
      class="AdminMenu__Link-t0dk93-2 kospmv"
    >
      Questions
    </div>
    <div
      class="AdminMenu__Link-t0dk93-2 kospmv"
    >
      Réponses
    </div>
    <div
      class="AdminMenu__Link-t0dk93-2 kospmv"
    >
      Unités
    </div>
    <div
      class="AdminMenu__Link-t0dk93-2 kospmv"
    >
      Utilisateurs
    </div>
    <h2
      class="Subtitle-sc-17ig8qv-0 AdminMenu__Subtitle-t0dk93-1 vshUS"
    >
      Maintenance
    </h2>
    <div
      class="AdminMenu__Link-t0dk93-2 kospmv"
    >
      Migrations
    </div>
  </div>
</div>
`;

exports[`[Contrib] layouts/<AdminMenu /> should match snapshot diff when the path has changed 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -1,16 +1,16 @@
  <DocumentFragment>
    <div
      class=\\"sc-bdVaJa sc-bwzfXH AdminMenu__Container-t0dk93-0 dYccBZ\\"
    >
      <div
-       class=\\"AdminMenu__Link-t0dk93-2 LnAQE\\"
+       class=\\"AdminMenu__Link-t0dk93-2 kospmv\\"
      >
        Tableau de bord
      </div>
      <div
-       class=\\"AdminMenu__Link-t0dk93-2 kospmv\\"
+       class=\\"AdminMenu__Link-t0dk93-2 LnAQE\\"
      >
        Conventions
      </div>
      <div
        class=\\"AdminMenu__Link-t0dk93-2 kospmv\\""
`;
